### PR TITLE
Ports Persistent Personal Photoalbums (PPPP) from TG

### PR DESCRIPTION
**Pull request by HanSolo1519**
## About The Pull Request
like the title says
This is black magic to me, I just squinted at the code both here and on TG and it seemed similar.

May god spare my soul

## Pre-Merge Checklist
- [ ] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [ ] You documented all of your changes.

## Changelog
:cl:
add: Added new things
/:cl:

### CODE REVIEW

File: code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -410,3 +410,31 @@
...
+	desc = "You carry your camera and personal photo album everywhere you go, and your scrapbooks are legendary among your coworkers."
+	value = 0
+	mob_trait = TRAIT_PHOTOGRAPHER
+	gain_text = span_notice("You know everything about photography.")
+	lose_text = span_danger("You forget how photo cameras work.")
+	medical_record_text = "Patient mentions photography as a stress-relieving hobby."
+
+/datum/quirk/item_quirk/photographer/add_unique(client/client_source)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/obj/item/storage/photo_album/personal/photo_album = new(get_turf(human_holder))
```
Comment (GremlingSS) [inline code/datums/traits/neutral.dm:425]: `var/obj/item/storage/photo_album/personal/photo_album`
to:
`var/obj/item/storage/photo_album/photo_album`

File: code/datums/traits/neutral.dm [hunk 2]
```diff
@@ -410,3 +410,31 @@
...
+	photo_album.persistence_id = "personal_[human_holder.last_mind?.key]" // this is a persistent album, the ID is tied to the account's key to avoid tampering
+	photo_album.persistence_load()
+	photo_album.name = "[human_holder.real_name]'s photo album"
+
+	give_item_to_holder(photo_album, list(LOCATION_BACKPACK = ITEM_SLOT_BACKPACK, LOCATION_HANDS = ITEM_SLOT_HANDS))
```
Comment (GremlingSS) [inline code/datums/traits/neutral.dm:430]:
```
give_item_to_holder(photo_album, list(LOCATION_BACKPACK = ITEM_SLOT_BACKPACK, LOCATION_HANDS = ITEM_SLOT_HANDS))
```
to:
```
if(!human_holder.equip_to_slot_if_possible(photo_album, SLOT_IN_BACKPACK, disable_warning = TRUE, bypass_equip_delay_self = TRUE)
    human_holder.put_in_hands(photo_album)
```

File: code/datums/traits/neutral.dm [hunk 3]
```diff
@@ -410,3 +410,31 @@
...
+	give_item_to_holder(
+		/obj/item/camera,
+		list(
+			LOCATION_NECK = ITEM_SLOT_NECK,
+			LOCATION_LPOCKET = ITEM_SLOT_LPOCKET,
+			LOCATION_RPOCKET = ITEM_SLOT_RPOCKET,
+			LOCATION_BACKPACK = ITEM_SLOT_BACKPACK,
+			LOCATION_HANDS = ITEM_SLOT_HANDS
+		)
+	)
```
Comment (GremlingSS) [inline code/datums/traits/neutral.dm:440]:
```
give_item_to_holder(
		/obj/item/camera,
		list(
			LOCATION_NECK = ITEM_SLOT_NECK,
			LOCATION_LPOCKET = ITEM_SLOT_LPOCKET,
			LOCATION_RPOCKET = ITEM_SLOT_RPOCKET,
			LOCATION_BACKPACK = ITEM_SLOT_BACKPACK,
			LOCATION_HANDS = ITEM_SLOT_HANDS
		)
	)
```
To:
```
var/obj/item/camera/cam = new(get_turf(human_holder))
if(!human_holder.equip_to_slot_if_possible(cam , SLOT_IN_BACKPACK, disable_warning = TRUE, bypass_equip_delay_self = TRUE)
    human_holder.put_in_hands(cam)
```

**Review by GremlingSS — CHANGES_REQUESTED** [review]
ae

**Review by Superlagg — APPROVED** [review]

**Review by GremlingSS — APPROVED** [review]
okidoki

**Pull request merged**